ecdsa.go

```diff
@@ -0,0 +1,157 @@
+package cose
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"errors"
+	"fmt"
+	"io"
+	"math/big"
+
+	"golang.org/x/crypto/cryptobyte"
+	"golang.org/x/crypto/cryptobyte/asn1"
+)
+
+// I2OSP - Integer-to-Octet-String primitive converts a nonnegative integer to
+// an octet string of a specified length `len(buf)`, and stores it in `buf`.
+// I2OSP is used for encoding ECDSA signature (r, s) into byte strings.
+//
+// Reference: https://datatracker.ietf.org/doc/html/rfc8017#section-4.1
+func I2OSP(x *big.Int, buf []byte) error {
+	if x.Sign() < 0 {
+		return errors.New("I2OSP: negative integer")
+	}
+	if len(x.Bits()) > len(buf) {
```

Comment:
RFC8017 Section 4.1 cases says:
> If x >= 256^xLen, output "integer too large" and stop.

`if len(x.Bits()) > len(buf) {` does not fully implement this check.

For example, the following invalid inputs are not catch by the if-check and result in a runtime panic inside `FillBytes`:

```go
cose.I2OSP(big.NewInt(256*256*256), make([]byte, 3))
```

I think this should do the trick:
```suggestion
	if (x.BitLen()+7)/8 > len(buf) {
```

Reply:
Good catch.